qemu/tests/qemu_guest_agent.py

```diff
@@ -986,6 +984,125 @@ def setup(self, test, params, env):
+        """
+        logging.info("Write time stamp to guest file per second as a backgroud job.")
+        fswrite_cmd = utils_misc.set_winutils_letter(
+            self.session, self.params["gagent_fs_test_cmd"])
+        self.fswrite_thread = utils_misc.InterruptedThread(self.session.cmd,
+                                                           (fswrite_cmd,
+                                                            (freeze_timeout * 2)))
+        self.fswrite_thread.start()
+        return self.fswrite_thread
+
+    @error_context.context_aware
+    def _fswrite_check(self):
+        """
+        get the result from vm and check if the time stamp is not continuous
+        if not continuous,guest file is frozen.
+        """
+        logging.info("Check the time stamp written result.")
+        process.system("rm -rf /tmp/fsfreeze.txt")
+        self.vm.copy_files_from("C:\\fsfreeze.txt", "/tmp/")
+        with open('/tmp/fsfreeze.txt', 'r') as f:
+            list_time = f.readlines()
+
+        for i in xrange(0, len(list_time)):
+            list_time[i] = list_time[i].rstrip('\r\r\n')
+
+        for i in xrange(1, len(list_time)):
+            num_d = float(list_time[i]) - float(list_time[i - 1])
+            if num_d > 2:
+                logging.info("Time stamp is not continuous, so the FS is frozen.")
+                return "frozen"
+
+        return "thaw"
+
+    @error_context.context_aware
+    def _action_before(self, *args):
+        session = self._get_session(self.params, None)
+        self.session = session
+        self._open_session_list.append(session)
+
+        error_context.context("Before freeze/thaw the FS, run the backgroud job.",
+                              logging.info)
+        freeze_timeout = args[0]
+        self._fswrite_bg_test(freeze_timeout)
+
+    @error_context.context_aware
+    def _action_after(self, *args):
+        flag = args[0]
+        error_context.context("Verify FS is %s within freeze time stamp" % flag,
+                              logging.info)
+        if self.fswrite_thread:
+            freeze_timeout = args[1]
+            logging.info("Waiting %s, then finish writing time stamp in "
+                         "guest file." % freeze_timeout)
+            self.fswrite_thread.join(freeze_timeout)
+            session2 = self._get_session(self.params, None)
+            self._open_session_list.append(session2)
+            session2.cmd("wmic process where name='python.exe' call terminate")
+
+        fs_status = self._fswrite_check()
+        if fs_status == flag:
+            logging.info("FS is %s as expected." % flag)
+        else:
+            self.test.fail("FS is not %s, it's %s." % (flag, fs_status))
+
+    @error_context.context_aware
+    def gagent_check_fsfreeze(self, test, params, env):
+        """
+        Test guest agent commands "guest-fsfreeze-freeze/status/thaw"
+        for windows guest
+
+        Test steps:
+        1) Check the FS is thawed.
```

Comment:
Can we add a step to check if the guest agent is running?

Reply:
there is no need as those check will be done in setup process.